### ISSUE TITLE
Getting error "Unable to find Bond for IP Address" with proper IP address and Bond Token configured

### ISSUE THREAD

**Issue by Nevs123** [aarons22/homebridge-bond#229]
**Describe the bug**
I am receiving a "Unable to find Bond for IP Address" error in my logs for Homebridge using a newly purchased BD-1000. I've tried everything I can think of (outlined below) and cannot get the bond to be picked up by Homebridge.

- Bond was successfully set up through the app, and I joined a ceiling fan to the bond
- The ceiling fan appears to work fine within the bond app

- I am running the following versions on Homebridge:
-- homebridge-config-ui-x: **v4.50.2**
-- Node.js version: **v18.12.1**

- IP address config in the bond app is set to static
- I ran the ping test on my terminal as recommended in the guide: **ping ZZFG32391.local**, this command returned the proper IP address that the app is showing (_192.168.2.61_)
- When configuring the IP address and Bond Token values in the plugin, I did this on my phone and copy/pasted the values into the proper fields to ensure no mistyping was done
- I've rebooted the home bridge server AND service multiple times since installing and configuring the plugin
- My home network configuration is very simple. Theres no segregated VLANs

**Information (please complete the following information):**
 - Plugin version: **homebridge-bond v3.2.10**
 - Bond firmware version: **v3.8.4**
 - How homebridge is run: **Homebridge UI**
 
**Logs**
Log output is very limited. I tried running this in debug mode but it did not provide ANY further detail:

[21/12/2022, 14:36:38] [Bond] Unable to find Bond for IP Address: 192.168.2.61. Skipping this Bond.
[21/12/2022, 14:36:38] [Bond] No valid Bonds available.

****** Bond Info *******
 bondId: **ZZFG32391**
 FW: **v3.8.4**
 API: v2 (_unsure how to check this_)
 Make: undefined
 Model: **BD-1000**
************************

**Comment by rajiviyengar** [comment]
I am facing the same error with my bond "Unable to find bond for IP address." I also copy and pasted the token to avoid mistyping. The bond bridge is set up using a static IP on my router. Pinging the bond is successful and bond is functioning normally via the app. Downgrading to version 3.2.9 did not solve the issue.

**Comment by Nevs123** [comment]
> I am facing the same error with my bond "Unable to find bond for IP address." I also copy and pasted the token to avoid mistyping. The bond bridge is set up using a static IP on my router. Pinging the bond is successful and bond is functioning normally via the app. Downgrading to version 3.2.9 did not solve the issue.

Glad to see I’m not the only one. I can even ping the IP address from the homebridge terminal and get a response back, but the plug-in for some reason is not able to recognize that the static IP is a Bond bridge.

Any guidance here or tests I should try? I’d like the keep the bond bridge but may send it back if I can’t solve this.

**Comment by rajiviyengar** [comment]
The bond bridge and homebridge plugin were working perfect for the past year. The error only started after my raspberry pi got corrupted, and I reinstalled homebridge and Raspbian.

**Comment by Nevs123** [comment]
I dug into the API documentation and ran some of the commands from the terminal within homebridge itself. Maybe some of this is useful for troubleshooting? I've redacted a few details for privacy.

**Token Command**
curl -i http://[BOND IP ADDRESS]/v2/token

```
HTTP/1.1 200 OK
BOND-Flags: 4
Content-Length: 214
Content-Type: application/json; charset=utf-8

{"locked":0,"pin_attempts_left":10,"token":"[token provided here]","nonce":"f50a964b8f2f74be","v1_nonce":"0000000000000000","account_code":"[Account Code Here]","v1_email":"","disable":false,"_":"bb32ff10","__":"bb32ff10"}
```

**Version Command**
curl -i http://[BOND IP ADDRESS]/v2/sys/version
```
HTTP/1.1 200 OK
BOND-Flags: 4
Content-Length: 275
Content-Type: application/json; charset=utf-8

{"target":"zermatt","fw_ver":"v3.8.4","fw_date":"Thu Nov 10 15:16:25 UTC 2022","uptime_s":998,"boot_patch":"not first boot","make":"Olibra","model":"BD-1000","branding_profile":"OLIBRA_BD1000","bondid":"[bond id here]","upgrade_http":true,"api":2,"_":"8a969a0a","__":"8a969a0a"}
```

**List Bond Devices Command**
I dont currently have any underlying devices configured

curl -H "BOND-Token: [BOND TOKEN HERE]" -i http://[BOND IP ADDRESS]/v2/devices

```
HTTP/1.1 200 OK
BOND-Flags: 4
Content-Length: 34
Content-Type: application/json; charset=utf-8

{"_":"00000000","__":"00000000"}
```

**Comment by patrickoleary1977** [comment]
Same issue here as well.

Multiple times per day, I need to reboot home bridge because any bond items in HomeKit stop working. When I check the log, it always shows the below. When I reboot, it fixes it 1/2 the time. Sometimes a 2nd reboot is necessary. When I check the bond app, the bond is available and I can control the devices from within Bond. I have the IP address of the device fixed for the bond. Thoughts?

Unable to find Bond for IP Address: 192.168.1.50. Skipping this Bond.
[1/19/2023, 3:00:24 AM] [homebridge-bond] No valid Bonds available.

**Comment by patrickoleary1977** [comment]
I think the issue with this is the Bond itself.  Anytime the Home app doesn't work, I check the Bond app and it says it is Syncing....

If it gets out of Syncing status, and if I reboot Homebridge, it all works again.  It just wont stay in Connected status.

**Comment by rajohn96** [comment]
anyone made any progress on this?

**Comment by cptoaks** [comment]
I have the same issue:
[4/19/2023, 4:15:31 PM] [Bond] Unable to find Bond for IP Address: x.x.x.x. Skipping this Bond.
[4/19/2023, 4:15:31 PM] [Bond] No valid Bonds available.
Also bond bridge shows 'status: syncing...' continuously in the bond app.

**Comment by patrickoleary1977** [comment]
I personally think it’s an issue with Bond. Even without Homebridge, I can’t get it to work now. On Jun 1, 2023, at 2:30 PM, 88gts ***@***.***> wrote:﻿
noticing this too, device is online and connected via app, pulling the IP and Token from the app and Homebridge reports the following each time, despite the host pinging the IP.
[6/1/2023, 11:24:40 AM] [Bond] Unable to find Bond for IP Address: 10.0.1.79. Skipping this Bond.
[6/1/2023, 11:24:40 AM] [Bond] No valid Bonds available.

**Comment by droopdog7** [comment]
Well, I am coming here to report the same issue.  Seems that there hasn't been a solution (sigh) but reviving this thread in case someone has it figured out.

**Comment by 88gts** [comment]
it is working for me, i noticed that the device needs to be online and powered up before Homebridge initializes it.

**Comment by droopdog7** [comment]
> it is working for me, i noticed that the device needs to be online and powered up before Homebridge initializes it.

Can you be a little more specific.  Also, I am not any kind of expert so go slow.

**Comment by 88gts** [comment]
your bond unit must have power and be online through the app when Homebridge starts up.

**Comment by droopdog7** [comment]
> your bond unit must have power and be online through the app when Homebridge starts up.

Thanks.  I've had the bond powered and online each time and still no joy.  Sigh.  Anyway, let me paste my config file in case someone sees an error.

{
    "bonds": [
        {
            "ip_address": "192.1XX.4.XX8",
            "token": "7XXXX947aXXXX2cd"
        }
    ],
    "include_dimmer": true,
    "include_toggle_state": true,
    "fan_speed_values": true,
    "_bridge": {
        "username": "0E:38:X4:48:X2:X1",
        "port": 48379
    },
    "platform": "Bond"
}

**Comment by dxdc** [comment]
I think I may have fixed the problem, which stems from:
- retry logic not functioning
- timeout kicking in

https://github.com/aarons22/homebridge-bond/pull/252

Curious to know if this is working for anyone else, it seems to have fixed my system.

**Comment by aarons22** [comment]
I published the fix from @dxdc into a beta release - https://github.com/aarons22/homebridge-bond/releases/tag/3.2.11-beta.1

I can't verify it myself, so if there is anyone here who can, that would be great :)

**Comment by y0m1g** [comment]
I faced this issue today (on 3.2.10); upgraded to [3.2.11-beta.2](https://github.com/aarons22/homebridge-bond/releases/tag/3.2.11-beta.2) and it's working fine again now! :)

**Comment by mosherc** [comment]
I don't know if this is exactly the same issue, but I can't get mine to work even on the new beta version. When I check my Netgear router, the bond bridge is not listed as an attached device (neither IP or mac address), but it is listed in access control. I set the IP address to the same as the one in access control in Netgear, in the iOS app. When I do `ping ZZHE...2.local` on my mac on the same network, I get cannot resolve unknown host, and of course an error in the homebridge logs. Anybody know what's up? The bridge works fine and connects over my 2.4ghz network fine.

**Comment by aarons22** [comment]
Fixed in [3.2.1](https://github.com/aarons22/homebridge-bond/releases/tag/3.2.11)
